utils/httpUtils.ts

```diff
@@ -72,3 +71,114 @@ export const notFound = (message: string) => {
     message: message,
   });
 };
+
+export const getFeatureFlagsFromRequest = context => {
+  return context.request.headers.get("x-should-read-from-nextgen");
+};
+
+export const shouldReadFromNextGen = async context => {
+  let shouldReadFromNextGen = getFeatureFlagsFromRequest(context);
+  if (shouldReadFromNextGen === true || shouldReadFromNextGen === "true" || shouldReadFromNextGen === "True") {
+    // if the header is set, return the value
+    return true;
+  }
+  return false;
+};
+
+export const formatFedQueryForNextGen = (query: string) => {
```

Comment:
any chance we could add a brief comment above about what this function is doing?  it's a bit hard to figure that out from reading the code.

Reply:
good idea. It is a very questionable little piece of code.